### PR TITLE
Only use comments after last reset

### LINKED ISSUE #1996: bodhi.server.models.Update.num_admin_approvals includes invalid comments
[```bodhi.server.models.Update.num_admin_approvals```](https://github.com/fedora-infra/bodhi/blob/3.0.0/bodhi/server/models.py#L2530-L2542) uses ```self.comments``` when it should use [```self.comments_since_karma_reset()```](https://github.com/fedora-infra/bodhi/blob/3.0.0/bodhi/server/models.py#L1210-L1230) instead.

This means that admin approvals are not cleared when developers change the Builds associated with an Update.

### PR DESCRIPTION
**Pull request by ghost**
fixes #1996

Signed-off-by: Vismay Golwala <vgolwala@redhat.com>

### CODE REVIEW

**Comment by vismay-golwala** [comment]
jenkies test

File: bodhi/server/models.py [hunk 1]
```diff
@@ -2396,7 +2397,8 @@ def get_testcase_karma(self, testcase):
             positive karma.
         """
         good, bad, seen = 0, 0, set()
-        for comment in reversed(self.comments):
+        # Reversing a generator object {self.comments_since_karma_reset}
+        for comment in reversed([_ for _ in self.comments_since_karma_reset]):
```
Comment (cverna) [inline bodhi/server/models.py]: see comment above

File: bodhi/server/models.py [hunk 2]
```diff
@@ -2846,7 +2848,8 @@ def __str__(self):
         val += u"""
   Submitter: %s
   Submitted: %s\n""" % (username, self.date_submitted)
-        if len(self.comments):
+        # Counting length of a generator object {self.comments_since_karma_reset}
+        if sum(1 for _ in self.comments_since_karma_reset):
```
Comment (cverna) [inline bodhi/server/models.py]: PEP8 recommends the following 
```
Yes: if not seq:
        if seq:

No: if len(seq):
       if not len(seq):
```
So we should drop the `if len()`/`if sum()` here. 

I would recommend converting `self.comments_since_karma_reset` into a list and store it in a variable (for example `comments_since_karma_reset`) then use this variable in the if statement.

File: bodhi/server/models.py [hunk 3]
```diff
@@ -2846,7 +2848,8 @@ def __str__(self):
...
             val += u"   Comments: "
             comments = []
             for comment in self.comments:
```
Comment (cverna) [inline bodhi/server/models.py]: I think we would want to use the `comments_since_karma_reset` list here

**Review by cverna — CHANGES_REQUESTED** [review]

File: bodhi/server/models.py [hunk 4]
```diff
@@ -2373,7 +2373,8 @@ def get_bug_karma(self, bug):
             positive karma.
         """
         good, bad, seen = 0, 0, set()
-        for comment in reversed(self.comments):
+        # Reversing a generator object {self.comments_since_karma_reset}
+        for comment in reversed([_ for _ in self.comments_since_karma_reset]):
```
Comment (cverna) [inline bodhi/server/models.py]: If you check `self.comments_since_karma_reset` is already reversing the order of the comments. So it should not be needed here.

https://github.com/fedora-infra/bodhi/blob/develop/bodhi/server/models.py#L1890

**Review by cverna — APPROVED** [review]
LGTM

**Pull request merged**